### PR TITLE
Update `inspec detect` to support APIs/Families

### LINKED ISSUE #2611: InSpec detect platform support
The `inspec detect` command currently caters to only operating systems. This command's wording needs to be changed to "platforms". The code should also be moved from the os.rb to platform.rb resource. 

We should update so family is taken from train so it works for both api/os. Release could also be used for api versions if we set them in train. Arch should hide if not set?

We should also wrap the detect block to not blow up with a train stack trace like shell/exec have.

### PR DESCRIPTION
**Pull request by jerryaldrichiii**
This closes #2611 

This does the following to `inspec detect`:
  - Modifies it to use the `platform` resource
  - Changes the output to mention Platform and show the family hierarchy
  - Changes the JSON output by changing `family` to `families`
  - Adds better error messaging (no more stacktraces!)
  - Adds support for APIs such as AWS/Azure
  - Hides Arch from API platforms output (not applicable)

### CODE REVIEW

**Review by miah — APPROVED** [review]

**Review by jquick — APPROVED** [review]
This looks good to me. Example output of the new detect:
screen shot 2018-02-14 at 8 31 24 am

File: lib/resources/platform.rb [hunk 1]
```diff
@@ -46,8 +50,19 @@ def in_family?(family)
...
-      @platform.family_hierarchy
+    def params
+      h = {
+        name: name,
+        families: families,
+      }
+
+      # Avoid adding Release/Arch for APIs
+      unless in_family?('api')
+        h[:release] = release
```
Comment (chris-rock) [inline lib/resources/platform.rb]: According to https://github.com/chef/inspec/issues/1661 section 5. The platform resource release is not optional. Do we have a sdk version that we could use for release of an api?

**Review by chris-rock — COMMENTED** [review]
Great improvement @jerryaldrichiii I added one comment.

**Comment by chris-rock** [comment]
We may want to add a schema to verify that the output conforms to our standard.

File: test/functional/inspec_detect_test.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    stdout = res.stdout
+    stdout.must_include "\n== Platform Details\n\n"
+    stdout.must_include "\nName:      \e[0;36m"
+    stdout.must_include "\nFamilies:  \e[0;36m"
+
+    stdout.wont_include "\nArch:"
+    stdout.wont_include "\nRelease:"
+  end
+end
```
Comment (chris-rock) [inline test/functional/inspec_detect_test.rb]: Can we add a json test for aws and azure?

File: lib/resources/platform.rb (hunk 1 above)
Comment (jquick) [reply lib/resources/platform.rb]: I can throw in a train PR to add the sdk version as the release for the transports then it will be available here.

Comment (jerryaldrichiii) [reply lib/resources/platform.rb]: Cool beans, thanks @jquick. I'll move release out of the omit category.

File: test/functional/inspec_detect_test.rb (hunk 2 above)
Comment (jerryaldrichiii) [reply test/functional/inspec_detect_test.rb]: Sure can. Latest update has tests for API + JSON. Let me know what you think.

File: lib/resources/platform.rb (hunk 1 above)
Comment (jquick) [reply lib/resources/platform.rb]: Fields added via https://github.com/chef/train/pull/258

**Comment by jquick** [comment]
apis now shows release:
screen shot 2018-02-14 at 1 30 28 pm

**Review by chris-rock — APPROVED** [review]
Thank you @jerryaldrichiii

**Pull request merged**